src/Grpc.Net.Client/Internal/OperatingSystem.cs

```diff
@@ -23,16 +23,23 @@ namespace Grpc.Net.Client.Internal;
 internal interface IOperatingSystem
 {
     bool IsBrowser { get; }
+    bool IsAndroid { get; }
 }
 
-internal class OperatingSystem : IOperatingSystem
+internal sealed class OperatingSystem : IOperatingSystem
 {
     public static readonly OperatingSystem Instance = new OperatingSystem();
 
     public bool IsBrowser { get; }
+    public bool IsAndroid { get; }
```

Comment:
Will this kind of problem only ever exists on Android or may there be other devices (in the future) be too?

So maybe this should be named
```suggestion
    public bool HasNativeIncompatibleHttpHandler { get; }
```
to be able to cover more devices w/o adding more properties here?

Reply:
Maybe. But because it’s an internal type we can rename it in the future.